src/components/InternalLink.tsx

```diff
@@ -8,8 +7,13 @@ import { parseInternalHref } from '../private/url';
 
 import * as styles from './InternalLink.css';
 
-interface Props<S = LocationState>
-  extends Omit<ComponentProps<typeof NavHashLink>, 'scroll' | 'smooth' | 'to'> {
+interface Props<S = any>
+  extends Omit<
+    ComponentProps<typeof NavHashLink>,
+    'className' | 'scroll' | 'smooth' | 'to'
+  > {
+  activeClassName?: ClassValue;
+  className?: ClassValue;
```

Comment:
This preserves the old `activeClassName` behaviour; the downside to this interface is you can't easily remove a `className` when active.

Given this is already a breaking change, maybe we should drop `activeClassName` and propagate the `className: (isActive: boolean) => ClassValue` function instead?